tools/GenerateNewControlProject.ps1

```diff
@@ -107,4 +107,75 @@ $xml.Save($testAppProject)
 FindAndReplaceInFile ($muxControlsDir + "\dev\Telemetry\RuntimeProfiler.h") "(\s*ProfId_Size.*\s*})" @"
 
         ProfId_$controlName,`$1
-"@
\ No newline at end of file
+"@
+# Randomize class name for multiple use of class in one powershell
+$id = get-random
+
+# We need double backslash for C# strings below
+$cleanMuxControlsDir = $muxControlsDir.Replace("\","\\") + "\\"
+
+echo "$cleanMuxControlsDir"
+
+$assemblies=(
+	"System","EnvDTE","EnvDTE80"
+)
+
+$source="
+using System;
+using EnvDTE;
+using EnvDTE80;
+using System.Collections.Generic;
+namespace SolutionHelper
+{
+	public static class SolutionRegister$id{
+		public static void Main(){
+            Console.WriteLine(`"Started script`");
+            Type dteType = null;
+            try
+            {
+                dteType = Type.GetTypeFromProgID(`"VisualStudio.DTE.16.0`", true);
+            } catch(Exception)
```

Comment:
According to [the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.type.gettypefromprogid?view=netframework-4.8), GetTypeFromProgID does not throw if it fails to retrieve the type; it just returns null.  Recommend checking against null rather than catching an exception.

Reply:
Whoops, never mind on this - I missed that you had passed in the parameter to make it throw.  Out of curiosity, why not just check for null?

Reply:
There was no real reason, I've changed it to a null check now.